### PR TITLE
#37 Horizontal list of meal card

### LINKED ISSUE #37: Horizontal list of meal card.
### Issuer:
Hung Ngoos
### Assignee(s):
Vy Nguyen

### Requirements:
image

As you have developed meal card, this time, you will try to combine those cards into a horizontal list. Furthermore, as you can see, there are three horizontal list stacking vertically in the pictures. You don't have to care about the colors at first. 
Try to design it with props not hard coding the mock data inside the components. 
Specifically,
I will input an object like this 
```javascript
{
    resultsType1 : [
        {
            title: "Meal 1",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category1"
            infoLink: "https://www.google.com"
        },
        {
            title: "Meal 2",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category1"
            infoLink: "https://www.google.com"
        },
        {
            title: "Meal 3",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category1"
            infoLink: "https://www.google.com"
        }
    ]

    resultType2 : [
        {
            title: "Meal 4",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category2"
            infoLink: "https://www.google.com"
        },
        {
            title: "Meal 5",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category2"
            infoLink: "https://www.google.com"
        },
        {
            title: "Meal 6",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category2"
            infoLink: "https://www.google.com"
        }
    ]

    resultType3 : [
        {
            title: "Meal 7",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category3"
            infoLink: "https://www.google.com"
        },
        {
            title: "Meal 8",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category3"
            infoLink: "https://www.google.com"
        },
        {
            title: "Meal 9",
            image: "https://source.unsplash.com/random/800x600",
            category: "Category3"
            infoLink: "https://www.google.com"
        }
    ]
}

```
There are three results type: type 1 for first row, type 2 for second, type 3 for third. Each of them has a list of meals, display them in that row. 
Each meal will have an infoLink. That is the link when we click the see more button of that card. 

NOTES: Please pulling main first, running npm run dev to see the current progress on frontend. Heading to ingredients page and you will see similar UI like this horizontal list of meal pages. 
You can use it as reference in the file IngredientsMain.tsx. 
This file is also displaying horizontal lists like this issue. 

### Dependencies:
-Please use Charka UI components: especially VStack and HStack. Search it on charka UI for more info. 

### Done requirements:
-Display as the picture provided above.

### PR DESCRIPTION
**Pull request by thuyvy119**
Close #37 Horizontal list of meal card
- Added ListofMeals component.
- Adjusted ImageCard component: added infoLink in See more button and set card variant to outline.
- Create a mock data function to render the Meal page.

https://github.com/hungngodev/Flavorie/assets/45311428/f81c4413-5ce4-4b22-9bef-837a92b4a7eb

### CODE REVIEW

File: client/src/components/meals/ListofMeals.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    {' '}
+                    <ImageCard
+                    imageProps={{
+                        src: meal.image,
+                        title: meal.title,
+                        category: meal.category,
+                        price: meal.price,
+                        infoLink: meal.infoLink,
+                    }}
+                    />
```
Comment (hungngodev) [inline client/src/components/meals/ListofMeals.tsx:91]: Please make the card smaller!

File: client/src/components/meals/ImageCard.tsx
Comment (hungngodev) [inline client/src/components/meals/ImageCard.tsx:1]: Please make a 1 line description for this one!

**Review by hungngodev — CHANGES_REQUESTED** [review]

File: client/src/components/meals/ImageCard.tsx
Comment (thuyvy119) [reply client/src/components/meals/ImageCard.tsx:1]: Updated list as below
https://github.com/hungngodev/Flavorie/assets/45311428/ff20571f-1268-4f6f-9bcb-a874886a2b2f

**Comment by thuyvy119** [comment]
Updated list as below

https://github.com/hungngodev/Flavorie/assets/45311428/32d2bbc2-94d8-4059-9f0a-f9c013f9cbaa

**Comment by hoatuyet423** [comment]
- Code look pretty good in general
- add more spacing between each horizontal block
- Make the tittle (for example "Meal 1", "Meal 2') more clear, either bold them or make them bigger, make sure that the seperation is clear for the user. Also this is personal but i feel like it would look better without the box around the tittle, since this page already have so many boxes .-.
- Color and the theme need to be change too but those can be decided later
- Otherwise everything look pretty good to me. Great works!

**Review by hungngodev — CHANGES_REQUESTED** [review]
Please fixing Jenny's comment by today!

**Comment by thuyvy119** [comment]
Thank you Jenny for your comment, already added more spacing and change font size for title of each meal type.

https://github.com/hungngodev/Flavorie/assets/45311428/2627d975-112f-470b-b075-287b1d0df117

**Review by hungngodev — APPROVED** [review]
Great work!!!

**Pull request merged**